### PR TITLE
Remove unnecessary `Array` -> `List` -> `Array` conversion in `ProcessBaseCommand.AllProcesses`

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

 * Removes redundant code from `ProcessBaseCommand.AllProcesses` left after functionality was removed in #4960.
 * Simplify and use coalesce compound assignment (like in #13396)
 * Update documentation comments for style.

## PR Context

Follow-up to #4960.

See also: #15053.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [NA] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [NA] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [NA] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [NA] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [NA] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [NA] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [NA] Issue filed: 
        - [NA] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [NA] Issue filed: 
        - [NA] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [NA] Issue filed: 
        - [NA] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [NA] Issue filed:

### CODE REVIEW

**Review by rjmholt — APPROVED** [review]

**Comment by rjmholt** [comment]
Waiting on review from @daxian-dbw or @adityapatwardhan

**Review by iSazonov — APPROVED** [review]

**Comment by xtqqczze** [comment]
@TravisEz13 Can you approve changes?

**Comment by TravisEz13** [comment]
@xtqqczze I'm not the reviewer for these changes.  @rjmholt  is the maintainer assigned to this.

**Pull request merged**
